core/src/com/unciv/models/ruleset/unique/UniqueTriggerActivation.kt

```diff
@@ -664,6 +664,29 @@ object UniqueTriggerActivation {
                 return true
             }
 
+            UniqueType.SellBuilding -> {
+
+                val applicableCities = civInfo.cities.asSequence().filter {
+                    it.matchesFilter(unique.params[1])
+                }
+
+                for (applicableCity in applicableCities) {
+                    var buildingsToSell = applicableCity.cityConstructions.getBuiltBuildings().filter {
+                        it.matchesFilter(unique.params[0])
```

Comment:
Why not apply both filters here? Filter for isSellable as well
And you don't need to .toSet() this at all

Reply:
I think you misunderstood me... .filter { A && B }

Reply:
I've updated this function according to your remark. Is this PR ready to be merged?